### ISSUE TITLE
highlight() doesn't work 

### ISSUE THREAD

**Issue by jasonbaik94** [ropensci/plotly#1289]
---
This is Jason Baik from Carnegie Mellon.
Hope you've been doing great.
When I try to replicate this line here 

`highlight(ggplotly(p), dynamic = T, persistent = T, selectize = T)`

from this presentation you gave at CMU, 

[https://talks.cpsievert.me/20180118/#22](url),

I can't highlight each of the lines separately. 
At first, I was instructed to download the development version of `ggplot2`. 
I did so, and ran the `highlight` function again, but it still doesn't highlight each of the lines separately.

Below, I will list my code:

```r
> dput(p)
structure(list(data = structure(list(anon_screen_name = c("0aaee8e27193a0db8bcd1594815e474e170956b8", 
"0aaee8e27193a0db8bcd1594815e474e170956b8", "0aaee8e27193a0db8bcd1594815e474e170956b8", 
"0aaee8e27193a0db8bcd1594815e474e170956b8", "0aaee8e27193a0db8bcd1594815e474e170956b8", 
"0aaee8e27193a0db8bcd1594815e474e170956b8", "0aaee8e27193a0db8bcd1594815e474e170956b8", 
"0aaee8e27193a0db8bcd1594815e474e170956b8", "0aaee8e27193a0db8bcd1594815e474e170956b8", 
"0aaee8e27193a0db8bcd1594815e474e170956b8", "0cc248e08127e057ac4248d532d6e72bdf577177", 
"0cc248e08127e057ac4248d532d6e72bdf577177", "0cc248e08127e057ac4248d532d6e72bdf577177", 
"0cc248e08127e057ac4248d532d6e72bdf577177", "0cc248e08127e057ac4248d532d6e72bdf577177", 
"0cc248e08127e057ac4248d532d6e72bdf577177", "0cc248e08127e057ac4248d532d6e72bdf577177", 
"0cc248e08127e057ac4248d532d6e72bdf577177", "0cc248e08127e057ac4248d532d6e72bdf577177", 
"0cc248e08127e057ac4248d532d6e72bdf577177", "17232046683105f3ed4c212f2843bf87d85019b8", 
"17232046683105f3ed4c212f2843bf87d85019b8", "17232046683105f3ed4c212f2843bf87d85019b8", 
"17232046683105f3ed4c212f2843bf87d85019b8", "17232046683105f3ed4c212f2843bf87d85019b8", 
"17232046683105f3ed4c212f2843bf87d85019b8", "17232046683105f3ed4c212f2843bf87d85019b8", 
"17232046683105f3ed4c212f2843bf87d85019b8", "17232046683105f3ed4c212f2843bf87d85019b8", 
"17232046683105f3ed4c212f2843bf87d85019b8", "288d6083da2553f56041212e805b2aafa5ec4ddf", 
"288d6083da2553f56041212e805b2aafa5ec4ddf", "288d6083da2553f56041212e805b2aafa5ec4ddf", 
"288d6083da2553f56041212e805b2aafa5ec4ddf", "288d6083da2553f56041212e805b2aafa5ec4ddf", 
"288d6083da2553f56041212e805b2aafa5ec4ddf", "288d6083da2553f56041212e805b2aafa5ec4ddf", 
"288d6083da2553f56041212e805b2aafa5ec4ddf", "288d6083da2553f56041212e805b2aafa5ec4ddf"
), latest_time = structure(c(1404165360, 1404165360, 1404165360, 
1404165360, 1404165360, 1404165360, 1404165360, 1404165360, 1404165360, 
1404165360, 1409781065, 1409781065, 1409781065, 1409781065, 1409781065, 
1409781065, 1409781065, 1409781065, 1409781065, 1409781065, 1410592786, 
1410592786, 1410592786, 1410592786, 1410592786, 1410592786, 1410592786, 
1410592786, 1410592786, 1410592786, 1404777469, 1404777469, 1404777469, 
1404777469, 1404777469, 1404777469, 1404777469, 1404777469, 1404777469
), class = c("POSIXct", "POSIXt"), tzone = "UTC"), platform = c("OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX"), course = c("Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014"
), week = c(1L, 2L, 3L, 4L, 5L, 6L, 7L, 8L, 9L, 10L, 1L, 2L, 
3L, 4L, 5L, 6L, 7L, 8L, 9L, 10L, 1L, 2L, 3L, 4L, 5L, 6L, 7L, 
8L, 9L, 10L, 1L, 2L, 4L, 5L, 6L, 7L, 8L, 9L, 10L), effort_sec = c(16302L, 
2199L, 25219L, 11310L, 9389L, 9990L, 4995L, 13796L, 12256L, 679L, 
17475L, 12961L, 9111L, 11206L, 2408L, 20861L, 18006L, 14763L, 
17910L, 16727L, 550L, 29672L, 26144L, 19915L, 14263L, 28795L, 
21092L, 24117L, 107683L, 30661L, 1066L, 3727L, 23834L, 19307L, 
35565L, 18767L, 12398L, 21363L, 28660L)), row.names = c(NA, -39L
), class = c("tbl_df", "tbl", "data.frame")), layers = list(<environment>), 
    scales = <environment>, mapping = structure(list(x = ~week, 
        y = ~effort_sec, group = ~anon_screen_name), class = "uneval"), 
    theme = list(), coordinates = <environment>, facet = <environment>, 
    plot_env = <environment>, labels = list(title = "Log-Transformed Effort over Time", 
        subtitle = "Took out the 14 sample points for Week 11", 
        x = "Week", y = "Log-Transformed Effort", group = "anon_screen_name")), class = c("gg", 
"ggplot"))


# This part here is the problem:
highlight(ggplotly(p), dynamic = T, persistent = T, selectize = T)
```

Thanks,

**Comment by cpsievert** [comment]
Hi @jasonbaik94, I'll need more information, specifically, how are you creating the obejct `p` with your data set?

**Comment by jasonbaik94** [comment]
Did you want to see the data manipulation / ggplot2 code inside the object `p`

**Comment by cpsievert** [comment]
Yes, the slide you linked to works...

You are using a different data set -- seeing how you created `p` would be helpful

**Comment by jasonbaik94** [comment]
Okay, here is the new `dput`:

`> dput(p)
structure(list(data = structure(list(anon_screen_name = c("0aaee8e27193a0db8bcd1594815e474e170956b8", 
"0aaee8e27193a0db8bcd1594815e474e170956b8", "0aaee8e27193a0db8bcd1594815e474e170956b8", 
"0aaee8e27193a0db8bcd1594815e474e170956b8", "0aaee8e27193a0db8bcd1594815e474e170956b8", 
"0aaee8e27193a0db8bcd1594815e474e170956b8", "0aaee8e27193a0db8bcd1594815e474e170956b8", 
"0aaee8e27193a0db8bcd1594815e474e170956b8", "0aaee8e27193a0db8bcd1594815e474e170956b8", 
"0aaee8e27193a0db8bcd1594815e474e170956b8", "0cc248e08127e057ac4248d532d6e72bdf577177", 
"0cc248e08127e057ac4248d532d6e72bdf577177", "0cc248e08127e057ac4248d532d6e72bdf577177", 
"0cc248e08127e057ac4248d532d6e72bdf577177", "0cc248e08127e057ac4248d532d6e72bdf577177", 
"0cc248e08127e057ac4248d532d6e72bdf577177", "0cc248e08127e057ac4248d532d6e72bdf577177", 
"0cc248e08127e057ac4248d532d6e72bdf577177", "0cc248e08127e057ac4248d532d6e72bdf577177", 
"0cc248e08127e057ac4248d532d6e72bdf577177", "17232046683105f3ed4c212f2843bf87d85019b8", 
"17232046683105f3ed4c212f2843bf87d85019b8", "17232046683105f3ed4c212f2843bf87d85019b8", 
"17232046683105f3ed4c212f2843bf87d85019b8", "17232046683105f3ed4c212f2843bf87d85019b8", 
"17232046683105f3ed4c212f2843bf87d85019b8", "17232046683105f3ed4c212f2843bf87d85019b8", 
"17232046683105f3ed4c212f2843bf87d85019b8", "17232046683105f3ed4c212f2843bf87d85019b8", 
"17232046683105f3ed4c212f2843bf87d85019b8"), latest_time = structure(c(1404165360, 
1404165360, 1404165360, 1404165360, 1404165360, 1404165360, 1404165360, 
1404165360, 1404165360, 1404165360, 1409781065, 1409781065, 1409781065, 
1409781065, 1409781065, 1409781065, 1409781065, 1409781065, 1409781065, 
1409781065, 1410592786, 1410592786, 1410592786, 1410592786, 1410592786, 
1410592786, 1410592786, 1410592786, 1410592786, 1410592786), class = c("POSIXct", 
"POSIXt"), tzone = "UTC"), platform = c("OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX", 
"OpenEdX", "OpenEdX", "OpenEdX", "OpenEdX"), course = c("Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014", "Medicine/MedStats/Summer2014", 
"Medicine/MedStats/Summer2014"), week = c(1L, 2L, 3L, 4L, 5L, 
6L, 7L, 8L, 9L, 10L, 1L, 2L, 3L, 4L, 5L, 6L, 7L, 8L, 9L, 10L, 
1L, 2L, 3L, 4L, 5L, 6L, 7L, 8L, 9L, 10L), effort_sec = c(16302L, 
2199L, 25219L, 11310L, 9389L, 9990L, 4995L, 13796L, 12256L, 679L, 
17475L, 12961L, 9111L, 11206L, 2408L, 20861L, 18006L, 14763L, 
17910L, 16727L, 550L, 29672L, 26144L, 19915L, 14263L, 28795L, 
21092L, 24117L, 107683L, 30661L)), row.names = c(NA, -30L), class = c("tbl_df", 
"tbl", "data.frame")), layers = list(<environment>), scales = <environment>, 
    mapping = structure(list(x = ~week, y = ~effort_sec, group = ~anon_screen_name), class = "uneval"), 
    theme = list(), coordinates = <environment>, facet = <environment>, 
    plot_env = <environment>, labels = list(title = "Log-Transformed Effort over Time", 
        subtitle = "Took out the 14 sample points for Week 11", 
        x = "Week", y = "Log-Transformed Effort", group = "anon_screen_name")), class = c("gg", 
"ggplot"))`

And the code I used to create `p`

`p <- fall_off_curve %>%
  slice(1:200) %>% 
  filter(week != 11,
         anon_screen_name %in%  c("0aaee8e27193a0db8bcd1594815e474e170956b8",
                                  "0cc248e08127e057ac4248d532d6e72bdf577177",
                                  "17232046683105f3ed4c212f2843bf87d85019b8",
                                  "288d6083da2553f56041212e805b2aafa5ec4ddf")) %>% 
  ggplot(aes(x = week, y = effort_sec, group = anon_screen_name)) +
  geom_line() +
  scale_x_continuous(labels = c("1", "2", "3", "4", "5", "6", "7",
                                "8", "9", "10", "11"),
                     breaks = seq(1, 11, 1)) +
  labs(x = "Week",
       y = "Log-Transformed Effort") +
  ggtitle("Log-Transformed Effort over Time",
          subtitle = "Took out the 14 sample points for Week 11")`


After running this:
`highlight(ggplotly(p), dynamic = T, persistent = T, selectize = T)`,

this is the output I get:

<img width="1140" alt="carson_highlight" src="https://user-images.githubusercontent.com/28586264/41369415-c5641528-6f12-11e8-9541-02f417f95504.png">



Basically, the problem with this graph is that when I click on a line, I can't highlight it. Also, 
the white box that allows me to select a certain value in the `group` doesn't appear.

**Comment by cpsievert** [comment]
`dput()` of the plot doesn't do much for me. I need `dput()` (or similar) of `fall_off_curve`

**Comment by jasonbaik94** [comment]
I can't release this data publicly, so I'll send you the `dput()` to your email

**Comment by cpsievert** [comment]
Looks like you haven't set the "primary key" for the interaction, this does the right thing for me:

```r
p <- fall_off_curve %>%slice(1:200) %>% filter(week != 11, anon_screen_name %in% c("0aaee8e27193a0db8bcd1594815e474e170956b8", "0cc248e08127e057ac4248d532d6e72bdf577177", "17232046683105f3ed4c212f2843bf87d85019b8", "288d6083da2553f56041212e805b2aafa5ec4ddf")) %>% crosstalk::SharedData$new(~anon_screen_name) %>% ggplot(aes(x = week, y = effort_sec, group = anon_screen_name)) + geom_line() + scale_x_continuous(labels = c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11"), breaks = seq(1, 11, 1)) + labs(x = "Week", y = "Log-Transformed Effort") + ggtitle("Log-Transformed Effort over Time", subtitle = "Took out the 14 sample points for Week 11")
```

**Comment by jasonbaik94** [comment]
Thank you so much!
